hwtracer/src/decode/ykpt/mod.rs

```diff
@@ -394,12 +394,13 @@ impl<'t> YkPTBlockIterator<'t> {
             }
             SuccessorKind::Dynamic => {
                 // We can only know the successor via a TIP update in a packet.
-                //
-                // FIXME: can't test without fixing indirect branch support for the
-                // block disambiguator pass in ykllvm.
-                //
-                // Test already written: tests/hwtracer_ykpt/indirect_jump.c
-                todo!();
+                self.seek_tip()?;
+                match self.cur_loc {
+                    ObjLoc::MainObj(off) => {
```

Comment:
I think the `return`s are superfluous (I think the block falls through to the function end)?

Reply:
I was following the precedent of the other cases.

It's a left-over from when this function was inlined into `do_next()`. I'll push a follow up commit that updates them all.

Reply:
Done in [eaea9da](https://github.com/ykjit/yk/pull/592/commits/eaea9da2c1fe1703b81f031fe8b0c4ef12a21e82)